src/CSS.cpp

```diff
@@ -13,36 +13,51 @@ namespace CSS
     {
         sProvider = gtk_css_provider_new();
 
-        struct CSSDir
+        std::vector<std::string> locations;
+        const char* home = std::getenv("HOME");
+        const char* env;
+
+        if ((env = std::getenv("XDG_CONFIG_HOME")) && strlen(env))
+        {
+            locations.push_back(env);
+        }
+        else if (home)
+        {
+            std::string s(home);
+            locations.push_back(s + "/.config");
+        }
+
+        if ((env = std::getenv("XDG_DATA_HOME")) && strlen(env))
+        {
+            locations.push_back(env);
+        }
+        else if (home)
         {
-            std::string xdgEnv;
-            std::string fallbackPath;
-            std::string relPath;
-        };
-        std::string home = getenv("HOME");
-        std::array<CSSDir, 4> locations = {
-            CSSDir{"XDG_CONFIG_HOME", home + "/.config",      "/gBar/style.css"}, // Local config
-            CSSDir{"XDG_DATA_HOME",   home + "/.local/share", "/gBar/style.css"}, // local user install
-            CSSDir{"",                "/usr/local/share",     "/gBar/style.css"}, // local all install
-            CSSDir{"",                "/usr/share",           "/gBar/style.css"}, // package manager all install
-        };
+            std::string s(home);
+            locations.push_back(s + "/.local/share");
+        }
+
+        if ((env = std::getenv("XDG_DATA_DIRS")))
```

Comment:
Hopefully one last thing: Missing strlen here (Like the ifs)
Other than that though: LGTM.